### DISCUSSION TITLE
Better way to detect the bot being disconnected?

### DISCUSSION THREAD

**Discussion by Notoric** [discord-jda/JDA#2535] [Questions and Help] [answered]
Currently I have a listener for onGuildVoiceUpdate that checks if the action was the bot leaving a channel, and then pulls the audit logs to check if someone was disconnected within 1 second either side of the action happening, but this has some issues and I'm sure there is a better method that I am just missing.
Any help is appreciated.

**Comment by MinnDevelopment** [comment]
With disconnected, do you mean kicked from the channel by a moderator? Or the bot leaving like `closeAudioConnection`? Or due to network issue?

What are you trying to achieve exactly? Reconnecting if the bot did not initiate a disconnect itself?

**Reply by Notoric** [reply]
Yes, by disconnect i mean when the bot is kicked from the voice channel by a moderator, I am trying to make this act as if the user had used /disconnect (Edit: This is a command I have created for the bot to make it leave the voice channel) to allow things to close down appropriately.

**Reply by MinnDevelopment** [reply]
If the channel is set to null that means it was kicked or left on its own

**Reply by Notoric** [reply]
Is there a way to differentiate between the bot leaving voluntarily or being kicked? other than checking the audit logs

**Reply by MinnDevelopment** [reply, marked as answer]
Not directly, no. However, you can also listen for the [GuildAuditLogEntryCreateEvent](https://docs.jda.wiki/net/dv8tion/jda/api/events/guild/GuildAuditLogEntryCreateEvent.html).

**Reply by Notoric** [reply]
Alright, thanks for responding, The audit log has an issue where if the same moderator has disconnected multiple users, the audit logs "stack" and it only says "Admin disconnected 2 users from voice". It also doesn't contain the member who was kicked from the voice channel, leading to potential edge cases where someone uses the /disconnect command at the same time as another user is kicked and the bot alerts a false kick.
